ocs_ci/deployment/disconnected.py

```diff
@@ -275,18 +340,32 @@ def mirror_index_image_via_oc_mirror(index_image, packages, icsp=None):
     cmd = (
         f"oc mirror --config {imageset_config_file} "
         f"docker://{config.DEPLOYMENT['mirror_registry']} "
-        "--dest-skip-tls --ignore-history"
+        " --dest-skip-tls --ignore-history"
     )
     if icsp:
         cmd += " --continue-on-error --skip-missing"
+    if idms:
+        # updating the config file to work with oc-mirror v2
+        with open(imageset_config_file, "r") as f:
+            lines = f.readlines()
+            lines[0] = "apiVersion: mirror.openshift.io/v2alpha1\n"
+            lines.index("storageConfig:\n")
+            lines = lines[:14]
+        with open(imageset_config_file, "w") as f:
+            f.writelines(lines)
+        cmd = (
+            f"oc mirror --config {imageset_config_file} "
+            f"docker://{config.DEPLOYMENT['mirror_registry']} "
+            "--workspace file://oc-mirror-workspace/results-files --v2"
+        )
```

Comment:
The usage of `oc-mirror --v2` shouldn't be based on the `idms` variable, but based on OCP version. The `icsp`/`idms` variables are used for unreleased versions, but we want to use `oc mirror --v2` also for GA version of ODF since 4.18.

Also I would propose to have two variants of the ImageSetConfig file template based on the version instead of updating the file here, because for example the file is printed to logs few lines above (IIRC as part of `templating.dump_data_to_temp_yaml(imageset_config_data, imageset_config_file)`) but when you change it here, it didn't correspond to the actual content used during the mirroring which might be misleading.